src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/letter/InternalReviewHomeOfficeResponseLetterGenerator.java

```diff
@@ -0,0 +1,81 @@
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AsylumCaseDefinition.APPEAL_REVIEW_OUTCOME;
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AsylumCaseDefinition.NOTIFICATION_ATTACHMENT_DOCUMENTS;
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.utils.AsylumCaseUtils.isAcceleratedDetainedAppeal;
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.utils.AsylumCaseUtils.isInternalCase;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AppealReviewOutcome;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.DocumentTag;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.CaseDetails;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.field.Document;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentCreator;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentHandler;
+
+@Component
+public class InternalReviewHomeOfficeResponseLetterGenerator implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final DocumentCreator<AsylumCase> internalReviewHomeOfficeResponseLetterCreator;
+    private final DocumentHandler documentHandler;
+
+    public InternalReviewHomeOfficeResponseLetterGenerator(
+            @Qualifier("internalReviewHomeOfficeResponseMaintainedLetter") DocumentCreator<AsylumCase> internalReviewHomeOfficeResponseLetterCreator,
+            DocumentHandler documentHandler
+    ) {
+        this.internalReviewHomeOfficeResponseLetterCreator = internalReviewHomeOfficeResponseLetterCreator;
+        this.documentHandler = documentHandler;
+    }
+
+    public boolean canHandle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        return callback.getEvent() == Event.REQUEST_RESPONSE_REVIEW
+                && callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && isInternalCase(asylumCase)
+                && !isAcceleratedDetainedAppeal(asylumCase)
+                && getAppealReviewOutcome(asylumCase).equals("decisionMaintained");
```

Comment:
do we need an "isappellantindetention" check here? otherwise this will trigger for non detained appeals too i think

Reply:
done